src/queries/role.ts

```diff
@@ -11,20 +13,37 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+type Role = AccessibilityRole | TextMatch;
+
+type ByRoleOptions = {
+  name?: TextMatch;
+};
+
+const filterByAccessibleName = (
+  node: ReactTestInstance,
+  name: TextMatch | undefined
+) => {
+  if (!name) return true;
```

Comment:
Previous check was also triggered by empty string.

```suggestion
  if (name == null) return true;
```